src/sagemaker_containers/transformer.py

```diff
@@ -0,0 +1,162 @@
+
+    Returns:
+        (obj) the loaded model.
+    """
+    raise NotImplementedError(textwrap.dedent("""
+    Please provide a model_fn implementation.
+    See documentation for model_fn at https://github.com/aws/sagemaker-python-sdk
+    """))
+
+
+def default_input_fn(input_data, content_type):
+    """Takes request data and de-serializes the data into an object for prediction.
+
+        When an InvokeEndpoint operation is made against an Endpoint running SageMaker model server,
+        the model server receives two pieces of information:
+
+            - The request Content-Type, for example "application/json"
+            - The request data content, which is at most 5 MB (5 * 1024 * 1024 bytes) in size.
+
+        The input_fn is responsible to take the request data and pre-process it before prediction.
+
+    Args:
+        input_data (obj): the request data content.
+        content_type (str): the request Content-Type.
+
+    Returns:
+        (obj): data ready for prediction.
+    """
+    return encoders.default_decoder.decode(input_data, content_type)
+
+
+def default_predict_fn(model, data):
+    """Function responsible for model predictions.
+
+    Args:
+        model (obj): model loaded by model_fn
+        data: de-serializes data returned by input_fn
+
+     Returns:
+         (obj): data ready for prediction.
+    """
+    raise NotImplementedError()
+
+
+def default_output_fn(prediction, accept):
+    """Function responsible to serialize the prediction for the response.
+
+    Args:
+        prediction (obj): prediction returned by predict_fn .
+        accept (str): accept content-type expected by the client.
+
+    Returns:
+        (worker.Response): a Flask response object with the following args:
+
+            * Args:
+                response: the serialized data to return
+                accept: the content-type that the data was transformed to.
+    """
+    return worker.Response(encoders.default_encoder.encode(prediction, accept), accept)
+
+
+class ClientError(BaseException):
+    pass
+
+
+class Transformer(object):
+    """The Transformer is a proxy between the worker and the framework transformation functions.
+
+    It implements the default framework functions for serving.
+
+    Examples:
+    >>>import os
+    >>>from sagemaker_containers import env, modules, transformer
+    >>>import Keras
+    >>>serving_env = env.ServingEnv()
+    >>>
+    >>>def predict_fn(model, data):
+    >>>     return model.predict(data)
+    >>>
+    >>>def model_fn(model_dir):
+    >>>     return Keras.models.load_model(os.path.join(model_dir, 'minimlmodel'))
+    >>>
+    >>>transformer = transformer.Transformer(predict_fn=predict_fn, model_fn=model_fn)
+    >>>
+    >>>mod = modules.download_and_import(serving_env.module_dir, serving_env.module_name)
+    >>>transformer.load_user_fns(mod)
+    """
+
+    def __init__(self, model_fn=None, input_fn=None, predict_fn=None, output_fn=None, error_class=ClientError):
```

Comment:
Are we not allowing transform_fn to be loaded from the user module? 

If we are, shouldn't it be a constructor argument?

Reply:
Yeah... Maybe that is the solution here. I wanted to separate creating the FrameworkTransformer and extracting the user functions.

My intention is to optionally allow customers to do the following:

*Let's suppose the following user script*
```python
from sagemaker_chainer import worker 

def predict_fn(...):
    pass

if __name__ == '__main__':
    worker.from_functions(predict_fn=predict_fn).run()